### PR TITLE
Cloud runtime vuln describe

### PR DESCRIPTION
**Pull request by mohsinenar**

### CODE REVIEW

File: src/ostorlab/apis/scan_vulnz.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
+"""Lists the remote vulnz."""
```
Comment (mohsinenar) [inline src/ostorlab/apis/scan_vulnz.py:1]:
```suggestion
"""List remote scan vulnz."""
```

File: src/ostorlab/apis/vulnz_describe.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
+"""Lists the remote vulnz."""
```
Comment (mohsinenar) [inline src/ostorlab/apis/vulnz_describe.py:1]:
```suggestion
"""Fetch a single vuln."""
```

File: src/ostorlab/apis/vulnz_describe.py [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+import json
+from typing import Dict, Optional
+
+from ostorlab.apis import request
+
+
+class VulnDescribeAPIRequest(request.APIRequest):
+    """Lists the remote vulnz of a scan."""
```
Comment (mohsinenar) [inline src/ostorlab/apis/vulnz_describe.py:10]:
```suggestion
    """Fetch a single vuln ."""
```

File: src/ostorlab/apis/vulnz_describe.py [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+    def __init__(self, scan_id: int):
+        self._scan_id = scan_id
+
+    @property
+    def query(self) -> Optional[str]:
+        """Defines the query to list the vulnz.
```
Comment (mohsinenar) [inline src/ostorlab/apis/vulnz_describe.py:17]:
```suggestion
        """Defines the query to fetch a vuln.
```

File: src/ostorlab/cli/vulnz/describe/describe.py [hunk 5]
```diff
@@ -1,70 +1,29 @@
...
-        for v in vulnerabilities:
-            _print_vulnerability(v)
-        console.success('Vulnerabilities listed successfully.')
-
-
-
-
-
+    if isinstance(ctx.obj['runtime'], cloud_runtime.CloudRuntime) and scan_id is None:
+        raise click.BadParameter('You should provide --scan-id. when using cloud runtime')
```
Comment (3asm) [inline src/ostorlab/cli/vulnz/describe/describe.py:24]:
```suggestion
        raise click.BadParameter('Please provide --scan-id / -s when using the cloud runtime')
```

File: src/ostorlab/cli/vulnz/describe/describe.py [hunk 6]
```diff
@@ -1,70 +1,29 @@
...
+    if isinstance(ctx.obj['runtime'], local_runtime.LocalRuntime) and scan_id is None and vuln_id is None:
+        raise click.BadParameter('You should provide --scan-id. or --vuln-id when using local runtime')
```
Comment (3asm) [inline src/ostorlab/cli/vulnz/describe/describe.py:26]:
```suggestion
        raise click.BadParameter('Please provide --scan-id / -s or --vuln-id / -v when using local runtime')
```

File: src/ostorlab/runtimes/cloud/runtime.py [hunk 7]
```diff
@@ -106,28 +111,86 @@ def install(self) -> None:
...
             vulnz_list_table = []
             for vulnerability in vulnerabilities:
                 vulnz_list_table.append({
-                    'id': str(vulnerability['kb']['id']),
-                    'highestRiskRating': styles.style_risk(vulnerability['highestRiskRating'].upper()),
-                    'cvss_v3_vector': vulnerability['kb']['cvssV3Vector'],
-                    'title': vulnerability['kb']['title'],
-                    'short_description': markdown.Markdown(vulnerability['kb']['shortDescription']),
+                    'id': str(vulnerability['id']),
+                    'riskRating': styles.style_risk(vulnerability['detail']['riskRating'].upper()),
```
Comment (3asm) [inline src/ostorlab/runtimes/cloud/runtime.py:124]: why is this camel case and the the other are not?

File: src/ostorlab/runtimes/cloud/runtime.py [hunk 8]
```diff
@@ -106,28 +111,86 @@ def install(self) -> None:
...
+                    'cvss_v3_vector': vulnerability['detail']['cvssV3Vector'],
+                    'title': vulnerability['detail']['title'],
+                    'short_description': markdown.Markdown(vulnerability['detail']['shortDescription']),
                 })
             columns = {
                 'Id': 'id',
                 'Title': 'title',
-                'highest Risk Rating': 'highestRiskRating',
+                'Risk Rating': 'riskRating',
```
Comment (3asm) [inline src/ostorlab/runtimes/cloud/runtime.py:132]: same here.

File: src/ostorlab/runtimes/cloud/runtime.py [hunk 9]
```diff
@@ -106,28 +111,86 @@ def install(self) -> None:
...
                 'CVSS V3 Vector': 'cvss_v3_vector',
                 'Short Description': 'short_description',
             }
             title = f'Scan {scan_id}: Found {len(vulnz_list_table)} vulnerabilities.'
             console.table(columns=columns, data=vulnz_list_table, title=title)
+            has_next_page: bool = response['data']['scan']['vulnerabilities']['pageInfo']['hasNext']
+            num_pages = response['data']['scan']['vulnerabilities']['pageInfo']['numPages']
+            if has_next_page is True:
+                console.info('Show the next page?')
```
Comment (3asm) [inline src/ostorlab/runtimes/cloud/runtime.py:141]:
```suggestion
                console.info('Fetch next page?')
```

File: src/ostorlab/runtimes/cloud/runtime.py [hunk 10]
```diff
@@ -106,28 +111,86 @@ def install(self) -> None:
...
+             'risk_rating': styles.style_risk(vulnerability['customRiskRating'].upper()),
+             'cvss_v3_vector': vulnerability['detail']['cvssV3Vector'],
+             'title': vulnerability['detail']['title'],
+             'short_description': markdown.Markdown(vulnerability['detail']['shortDescription']),
+             }
+        ]
+        columns = {
+            'Id': 'id',
+            'Title': 'title',
+            'Risk rating': 'risk_rating',
```
Comment (3asm) [inline src/ostorlab/runtimes/cloud/runtime.py:164]:
```suggestion
            'Risk Rating': 'risk_rating',
```

Comment (3asm) [reply src/ostorlab/runtimes/cloud/runtime.py:164]: Be consist with the casing.

File: src/ostorlab/runtimes/cloud/runtime.py [hunk 11]
```diff
@@ -106,28 +111,86 @@ def install(self) -> None:
...
+            'CVSS V3 Vector': 'cvss_v3_vector',
```
Comment (3asm) [inline src/ostorlab/runtimes/cloud/runtime.py:165]:
```suggestion
            'CVSSv3 Vector': 'cvss_v3_vector',
```

File: src/ostorlab/runtimes/cloud/runtime.py [hunk 12]
```diff
@@ -106,28 +111,86 @@ def install(self) -> None:
...
+                response = api_runner.execute(
+                    scan_vulnz.ScanVulnzDescribeAPIRequest(scan_id=scan_id, vuln_id=vuln_id, page=page))
+                vulnerabilities = response['data']['scan']['vulnerabilities']['vulnerabilities']
+                for v in vulnerabilities:
+                    self._print_vulnerability(v)
+                num_pages = response['data']['scan']['vulnerabilities']['pageInfo']['numPages']
+                console.success(f'Vulnerabilities listed successfully. page {page} of {num_pages} pages')
+                has_next_page: bool = response['data']['scan']['vulnerabilities']['pageInfo']['hasNext']
+                if has_next_page is True:
+                    console.info('Show the next page?')
```
Comment (3asm) [inline src/ostorlab/runtimes/cloud/runtime.py:190]:
```suggestion
                    console.info('Fetch next page?')
```

File: src/ostorlab/runtimes/cloud/runtime.py [hunk 13]
```diff
@@ -106,28 +111,86 @@ def install(self) -> None:
...
+                    page = page + 1
+                    if click.confirm(f'page {page + 1} of {num_pages}'):
+                        self.describe_vuln(scan_id=scan_id, vuln_id=vuln_id, page=page,number_elements=number_elements)
+
+        except runner.ResponseError:
+            console.error(f'Scan with ID {scan_id} not found')
```
Comment (3asm) [inline src/ostorlab/runtimes/cloud/runtime.py:196]:
```suggestion
            console.error(f'Scan {scan_id} not found')
```

File: src/ostorlab/runtimes/local/runtime.py [hunk 14]
```diff
@@ -505,4 +506,47 @@ def list_vulnz(self, scan_id: int):
...
+             'risk_rating': styles.style_risk(vulnerability.risk_rating.value.upper()),
+             'cvss_v3_vector': vulnerability.cvss_v3_vector,
+             'title': vulnerability.title,
+             'short_description': markdown.Markdown(vulnerability.short_description),
+             }
+        ]
+        columns = {
+            'Id': 'id',
+            'Title': 'title',
+            'Risk rating': 'risk_rating',
```
Comment (3asm) [inline src/ostorlab/runtimes/local/runtime.py:527]: Please change the casing and CVSSv3 per previous comment.

**Review by 3asm — CHANGES_REQUESTED** [review]

File: setup.cfg [hunk 15]
```diff
@@ -60,6 +60,7 @@ install_requires =
     tenacity
     sqlalchemy
     semver
+    markdownify
```
[the remark this answers is not in the document]
Comment (mohsinenar) [reply setup.cfg:63]: Noted. I'll create a ticket.

**Review by 3asm — APPROVED** [review]

**Pull request merged**